ios/keyman/Keyman/Keyman/Classes/View/KMNavigationBarBackgroundView.swift

```diff
@@ -0,0 +1,84 @@
+//
+//  KMNavigationBarBackgroundView.swift
+//  Keyman
+//
+//  Created by Jacob Bullock on 4/26/18.
+//  Copyright © 2018 SIL International. All rights reserved.
+//
+
+import Foundation
+import UIKit
+
+class KMNavigationBarBackgroundView: UIView {
+
+    var imageLeadingConstraint: NSLayoutConstraint?
+    var logoImageView: UIImageView?
+
+    required public init?(coder: NSCoder) {
+        super.init(coder: coder)
+        self.commonInit()
+    }
+
+    public override init(frame: CGRect) {
+        super.init(frame: frame)
+        self.commonInit()
+    }
+
+    func commonInit() {
+        self.backgroundColor = .clear
+        self.isUserInteractionEnabled = false
+        setupLogo()
+        setupUnderscore()
+    }
+    
+    func hideLogo() {
+        if let logoImageView = logoImageView {
+            logoImageView.alpha = 0
+        }
+    }
+
+    func setupUnderscore() {
+        let v = KMUnderscoreView(frame: CGRect())
+        self.addSubview(v)
+        v.translatesAutoresizingMaskIntoConstraints = false
+
+        v.constrainEqually(to: self, attribute: .bottom).isActive  = true
+        v.constrainEqually(to: self, attribute: .leading).isActive  = true
+        v.constrainEqually(to: self, attribute: .trailing).isActive  = true
+        v.constrainEqually(to: 3, attribute: .height).isActive  = true
```

Comment:
We've got a number of magic constants floating through this code? Given we are doing it fresh, I'd like to see those pulled out and given friendly names.